langserver/langserver.py

```diff
@@ -151,8 +151,28 @@ def usages(script, parent_span):
                                           "Script.usages"):
             return script.usages()
 
+    def setenv_from_initialization_options(self, params):
+        '''Sets environment variables from initializationOptions. This sets
+        environment variables for the entire process. In the future, this
+        might be modified to manage environment variables per workspace.'''
+
+        if "initializationOptions" not in params:
+            return
+        initOptions = params["initializationOptions"]
+        if not isinstance(initOptions, dict):
+            return
+        if "env" not in initOptions:
+            return
+        envVars = initOptions["env"]
+        if not isinstance(envVars, dict):
+            return
+        for k, v in envVars.items():
+            os.environ[k] = v
```

Comment:
@beyang this means that if a user removes an environment variable from the config, it will remain in effect until the process is restarted. This will likely lead to very confusing behaviour to the user.